src/core_modules/capture-core/components/WidgetProfile/WidgetProfile.component.js

```diff
@@ -54,6 +58,17 @@ const WidgetProfilePlain = ({ teiId, programId, showEdit = false, orgUnitId = ''
             };
         }), [clientAttributesWithSubvalues]);
 
+    useEffect(() => {
+        hasError && setTeiModalState(TEI_MODAL_STATE.OPEN_ERROR);
+    }, [hasError]);
+
+    useEffect(() => {
+        if (storedAttributeValues?.length > 0) {
+            setTeiModalState(TEI_MODAL_STATE.CLOSE);
+            onUpdateTeiAttributeValues && onUpdateTeiAttributeValues(storedAttributeValues, teiDisplayName);
+        }
```

Comment:
We will need to do cleanup here. If not, values can/will be overwritten when we open another tei.